### PR TITLE
rec: delint rec-main.cc wrt high complexity

### PR DESCRIPTION
**Pull request by omoerbeek**
### Checklist

I have:
- [X] read the [CONTRIBUTING.md](https://github.com/PowerDNS/pdns/blob/master/CONTRIBUTING.md) document
- [x] compiled this code
- [x] tested this code
- [ ] included documentation (including possible behaviour changes)
- [ ] documented the code
- [ ] added or modified regression test(s)
- [ ] added or modified unit test(s)
- [ ]  checked that this code was merged to master

### CODE REVIEW

File: pdns/recursordist/rec-main.cc [hunk 1]
```diff
@@ -2216,194 +2218,201 @@ class PeriodicTask
   const string name;
 };
 
-static void houseKeeping(void*)
+static void houseKeeping0(Logr::log_t log)
```
Comment (zeha) [inline pdns/recursordist/rec-main.cc]: what does 0 mean here?

Comment (omoerbeek) [reply pdns/recursordist/rec-main.cc]: `houseKeeping0` is the body of the original main loop of `houseKeeping`.  I agree it could be named better.

File: .clang-tidy.full [hunk 2]
```diff
@@ -253,7 +253,7 @@ CheckOptions:
   - key:             modernize-pass-by-value.ValuesOnly
     value:           'false'
   - key:             readability-function-cognitive-complexity.IgnoreMacros
-    value:           'true'
+    value:           'false'
```
Comment (rgacogne) [inline .clang-tidy.full:256]: Is that change intended? Don't we want to keep that value set to `true` so it works as on recent enough version of clang-tidy?

File: pdns/recursordist/rec-main.cc [hunk 3]
```diff
@@ -1245,28 +1247,29 @@ void broadcastFunction(const pipefunc_t& func)
...
+  unsigned int thread = 0;
   for (const auto& threadInfo : RecThreadInfo::infos()) {
-    if (n++ == RecThreadInfo::id()) {
+    if (thread++ == RecThreadInfo::id()) {
       func(); // don't write to ourselves!
       continue;
     }
 
-    ThreadMSG* tmsg = new ThreadMSG();
+    ThreadMSG* tmsg = new ThreadMSG(); // NOLINT: manual onwership handling
```
Comment (rgacogne) [inline pdns/recursordist/rec-main.cc:1257]:
```suggestion
    ThreadMSG* tmsg = new ThreadMSG(); // NOLINT: manual ownership handling
```

File: pdns/recursordist/rec-main.cc [hunk 4]
```diff
@@ -1322,30 +1325,30 @@ T broadcastAccFunction(const std::function<T*()>& func)
...
 
     T* resp = nullptr;
-    if (read(tps.readFromThread, &resp, sizeof(resp)) != sizeof(resp))
+    if (read(tps.readFromThread, &resp, sizeof(resp)) != sizeof(resp)) // NOLINT: sizeof correct
       unixDie("read from thread pipe returned wrong size or error");
 
     if (resp) {
       ret += *resp;
-      delete resp;
+      delete resp; // NOLINT: manual ownershipm handling
```
Comment (rgacogne) [inline pdns/recursordist/rec-main.cc:1351]:
```suggestion
      delete resp; // NOLINT: manual ownership handling
```

File: pdns/recursordist/rec-main.cc [hunk 5]
```diff
@@ -1945,6 +1778,298 @@ static int serviceMain(int /* argc */, char* /* argv */[], Logr::log_t log)
...
+         log->error(Logr::Error, e.reason, "Cannot load Lua configuration"));
+    return 1;
+  }
+
+  parseACLs();
+  initPublicSuffixList(::arg()["public-suffix-list-file"]);
+
+  initDontQuery(log);
+
+  /* this needs to be done before parseACLs(), which call broadcastFunction() */
```
Comment (rgacogne) [inline pdns/recursordist/rec-main.cc:1918]: Either this is not true or we have a problem.

**Review by Habbie — APPROVED** [review]

File: .clang-tidy.full (hunk 2 above)
Comment (omoerbeek) [reply .clang-tidy.full:256]: This was not intended to be committed

File: pdns/recursordist/rec-main.cc (hunk 5 above)
Comment (omoerbeek) [reply pdns/recursordist/rec-main.cc:1918]: `parseACLs()` does call `broadcastFunction()`, but that has a special case for when the recursor threads have not been set up yet. So the comment is not relevant.

**Review by rgacogne — APPROVED** [review]
